src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -867,7 +868,7 @@ bool HasAnyProfileDataForInput()
                 "DispatchImports",
                 ReadyToRunImportSectionType.StubDispatch,
                 ReadyToRunImportSectionFlags.PCode,
-                (byte)Target.PointerSize,
+                this.OptimizationFlags.EnableCachedInterfaceDispatchSupport ? (byte)(2 * Target.PointerSize) : (byte)Target.PointerSize,
```

Comment:
What happens if you generate an R2R image with cached interface dispatch support so each entry is two pointers, but you load it with a runtime that was built without support or doesn't have it enabled? I assume it will fail to load the R2R image in a deterministic, non-crashy way?

I would have assumed that if the entry size is different it should be a different table

Reply:
We'll load the image, and code will actually work, but we won't be able to swap these entrypoints to cached interface dispatch stubs, instead we'll have to go down a slow path every time. It'll work, but perf will be terrible.